### PR TITLE
Generate forecast based on given coordinates

### PR DESCRIPTION
**Pull request by admiral96**

### CODE REVIEW

File: app/weather_parser.py [hunk 1]
```diff
@@ -5,27 +5,19 @@
...
-                lat=lat,
-                lng=lng,
-                time=time
-            ))
+        request = requests.get('{url}/{key}/{lat},{lng},{time}'.format(
+            url=url,
+            key=FORECAST_API_KEY,
+            lat=lat,
+            lng=lng,
+            time=time
```
Comment (mknecht) [inline app/weather_parser.py]: tc

File: app/geocode_api_parser.py [hunk 2]
```diff
@@ -33,6 +35,21 @@ def get_place_info(self):
         self.place_type = geocode_info['results'][0]['address_components'][0]['types'][0]
         return {'place_type': self.place_type, 'formatted_address': self.formatted_address}
 
+    def get_place_time(self):
+        coordinates = self.get_coordinates()
+        timezone_request_params = {
+            'location': '{},{}'.format(coordinates['lat'], coordinates['lng']),
```
Comment (mknecht) [inline app/geocode_api_parser.py]: @admiral96 how would you do this line with the string.join() function? :)

File: app/geocode_api_parser.py [hunk 3]
```diff
@@ -33,6 +35,21 @@ def get_place_info(self):
...
+            'timestamp': 1331161200,
+            'key': TIMEZONE_API_KEY
```
Comment (mknecht) [inline app/geocode_api_parser.py]: tc

File: app/views.py [hunk 4]
```diff
@@ -26,7 +26,7 @@ def show_user_input():
 @url.route('/weather_forecast')
 def generate_forecast():
     if not Geocode.check_geocode_api_is_present() and WeatherForecast.check_forecast_api_is_present():
```
Comment (mknecht) [inline app/views.py]: what about timezone key?

File: app/geocode_api_parser.py [hunk 5]
```diff
@@ -33,6 +35,21 @@ def get_place_info(self):
         self.place_type = geocode_info['results'][0]['address_components'][0]['types'][0]
         return {'place_type': self.place_type, 'formatted_address': self.formatted_address}
 
+    def get_place_time(self):
```
Comment (mknecht) [inline app/geocode_api_parser.py]: this should be a property

File: requirements.txt [hunk 6]
```diff
@@ -2,5 +2,6 @@ flake8==2.5.4
 Flask==0.10.1
 gunicorn==19.4.5
 pytest==2.8.7
+pytz==2016.3
```
Comment (mknecht) [inline requirements.txt]: 

File: app/geocode_api_parser.py [hunk 7]
```diff
@@ -33,6 +35,21 @@ def get_place_info(self):
...
+        coordinates = self.get_coordinates()
+        timezone_request_params = {
+            'location': '{},{}'.format(coordinates['lat'], coordinates['lng']),
+            'timestamp': 1331161200,
+            'key': TIMEZONE_API_KEY
+        }
+        timezone_request = json.loads(requests.get(
+            'https://maps.googleapis.com/maps/api/timezone/json?',
```
Comment (mknecht) [inline app/geocode_api_parser.py]: constant

File: app/geocode_api_parser.py [hunk 8]
```diff
@@ -33,6 +35,21 @@ def get_place_info(self):
...
+            params=timezone_request_params
+        ).text)
+        timezone = timezone_request['timeZoneId']
+        tz = pytz.timezone(timezone)
```
Comment (mknecht) [inline app/geocode_api_parser.py]: what if pytz does not know timezone from google

**Comment by mknecht** [comment]
test pleaaaaaaaaaaaaaase!

File: app/weather_parser.py (hunk 1 above)
Comment (admiral96) [inline app/weather_parser.py]: what's tc?

Comment (mknecht) [inline app/weather_parser.py]: trailing comma

File: app/geolocation_api_parser.py [hunk 9]
```diff
@@ -0,0 +1,50 @@
+import json
+import pytz
+import requests
+from config import GEOCODE_API_KEY, TIMEZONE_API_KEY
+from datetime import datetime
+
+
+geocode_url = 'https://maps.googleapis.com/maps/api/geocode/json?'
+timezone_url = 'https://maps.googleapis.com/maps/api/timezone/json?'
```
Comment (mknecht) [inline app/geolocation_api_parser.py]: like i said, these are constants and should therefore by convention be uppercase

File: test_views.py [hunk 10]
```diff
@@ -65,16 +65,17 @@ def test_invalid_weather_url_empty(self):
...
+    def test_invalid_weather_url_invalid_address(self, is_place_query_valid):
         is_place_query_valid.return_value = False
         result = self.client.get('/weather_forecast?address=pppppppppp')
         self.assertTrue(is_place_query_valid.called)
         self.assertEqual(result.status_code, 400)
 
     @patch('app.weather_parser.WeatherForecast.generate_forecast')
-    @patch('app.geocode_api_parser.Geocode.get_address_query')
-    def test_valid_weather_url_valid_address(self, get_address_query, generate_forecast):
+    @patch('app.geolocation_api_parser.Geolocation.get_local_time')
```
Comment (mknecht) [inline test_views.py]: nope, get_local_time is doing too much than to be patched. patch surgically, only the minimum.

File: app/geolocation_api_parser.py [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+
+class Geolocation:
+
+    def __init__(self, address):
+        self.address = address
+        self.coordinates = self.get_coordinates() if self.is_place_query_valid() else None
+        self.local_time = self.get_local_time() if self.is_place_query_valid() else None
```
Comment (mknecht) [inline app/geolocation_api_parser.py]: these attributes should be marked as private.

**Comment by admiral96** [comment]
Build on travis will now pass if this PR will be merged.

**Comment by admiral96** [comment]
Will close this PR, I'll just make another one. I accidentally removed the remote without knowing the consequences, my bad.

**Pull request closed without merging**